tutorial/lesson_22_jit_performance.cpp

```diff
@@ -0,0 +1,263 @@
+        const Target target = get_jit_target_from_environment();
+        pipeline.compile_jit(target);
+
+        size_t count = 0;
+        double t = benchmark(samples, iterations, [&]() {
+            Buffer<uint16_t> result(1024, 1024);
+            pipeline.realize(result);
+            ++count;
+        });
+ 
+        // On a MacBook Pro M1, this should be about the same as the previous run (about ~175000 times/sec)
+        //
+        // This may seem somewhat surprisingly, since compiling before realizing doesn't seem to make 
+        // much of a difference to the previous case.  However, the first call to realize() will implicitly
+        // JIT-compile and cache the generated code associated with the Pipeline object, which is basically 
+        // what we've done here. Each subsequent call to realize uses the cached version of the native code, 
+        // so there's no additional overhead, and the cost is amortized as we re-use the pipeline.
+        std::cout << "Execute Pipeline (compile before realize): " << int(count / t) << " times/sec\n";
+
+
+        // Another subtlety is the creation of the result buffer ... the declaration implicitly
+        // allocates memory. Instead, we could use the realize({1024, 1024}) call which use the
+        // buffer managed by the pipeline object for the outputs ...
+        count = 0;
+        t = benchmark(samples, iterations, [&]() {
+            Buffer<uint16_t> result = pipeline.realize({1024, 1024});
+            ++count;
+        });
+
+        // On a MacBook Pro M1, this should be about the same as the previous run (about ~175000 times/sec).
+        std::cout << "Execute Pipeline (same but with realize({})): " << int(count / t) << " times/sec\n";
+
+        // Or ... we could move the declaration of the result buffer outside the timing loop, and
+        // re-use the allocation (with the caveat that we will be stomping over its contents on each 
+        // execution).
+        Buffer<uint16_t> result(1024, 1024);
+
+        count = 0;
+        t = benchmark(samples, iterations, [&]() {
+            pipeline.realize(result);
+            ++count;
+        });
+
+        // On a MacBook Pro M1, this should be much more efficient ... ~200000 times/sec (or 10-12% faster).
+        std::cout << "Execute Pipeline (re-use buffer with realize): " << int(count / t) << " times/sec\n";
+
+    }
+
+    // Alternatively, we could compile to a Callable object ...
+    {
+        Pipeline pipeline = make_pipeline();
+        const Target target = get_jit_target_from_environment();
+
+        // Here, we can ask the pipeline for its argument list (these are either Params,
+        // ImageParams, or Buffers) so that we can construct a Callable object with the same 
+        // calling convention.
+        auto arguments = pipeline.infer_arguments();
+
+        // The Callable object acts as a convienient way of invoking the compiled code like
```

Comment:
convenient

(side note: it's 2023, why don't we have smart spellcheckers that are useful for typos in code comments?)